lib/SILGen/SILGenDestructor.cpp

```diff
@@ -31,15 +31,60 @@ void SILGenFunction::emitDestroyingDestructor(DestructorDecl *dd) {
     Loc.markAutoGenerated();
 
   auto cd = cast<ClassDecl>(dd->getDeclContext());
+  auto &C = cd->getASTContext();
   SILValue selfValue = emitSelfDecl(dd->getImplicitSelfDecl());
 
   // Create a basic block to jump to for the implicit destruction behavior
   // of releasing the elements and calling the superclass destructor.
   // We won't actually emit the block until we finish with the destructor body.
   prepareEpilog(None, false, CleanupLocation(Loc));
 
+  auto cleanupLoc = CleanupLocation(Loc);
+
+  SILBasicBlock *deinitBodyBB = nullptr;
+  SILBasicBlock *finishBB = nullptr;
+  if (cd->isDistributedActor()) {
+    auto remoteBB = createBasicBlock("remoteActorDeinitBB");
+    finishBB = createBasicBlock("finishDeinitBB");
+    deinitBodyBB = createBasicBlock("deinitBodyBB");
+
+    // FIXME: what should the type of management be for this?
+    auto managedSelf = ManagedValue::forBorrowedRValue(selfValue);
+
+    auto selfTy = F.mapTypeIntoContext(cd->getDeclaredInterfaceType());
+    emitDistributedIfRemoteBranch(
```

Comment:
For isolated deinit I need this check to live inside deallocating deinit. Because isolation happens in deallocating deinit, and remote proxies don't need isolation. I would really appreciate if you could place the check already in the deallocating deinit, so that I don't need to move it later.

See [I would really appreciate that.](https://github.com/nickolas-pohilets/swift-evolution/blob/mpokhylets/isolated-deinit/proposals/NNNN-isolated-synchronous-deinit.md#interaction-with-distributed-actors)

Reply:
We're a bit tight on timing on getting changes into 5.7 betas -- I'm happy to do that in a follow up PR though.